plugins/modules/network_settings_workflow_manager.py

```diff
@@ -2040,37 +2129,48 @@ def get_have_reserve_pool(self, reserve_pool_details):
 
             # Check if the Reserved Pool exists in Cisco Catalyst Center
             # based on the provided name and site name
-            reserve_pool.append(self.reserve_pool_exists(name, site_name))
-            if not reserve_pool[reserve_pool_index].get("success"):
+            if self.reserve_pool_exists(name, site_name):
+                reserve_pool.append(self.reserve_pool_exists(name, site_name))
+            else:
+                self.have.update({"reservePool": reserve_pool})
+                return self
+
+            if name != "" and not reserve_pool[reserve_pool_index].get("success"):
+                return self.check_return_status()
```

Comment:
In few places, we are returning self and few places self.check_return_status()... Can you check?